db/versions/0023.yml

```diff
@@ -0,0 +1,241 @@
+version: 23
+methods:
+  get_index_namespaces:
+    description: |-
+      Get existing index_namespaces filtered by the optional arguments,
+      ordered by the `parent` and `name`.
+      If the pagination arguments are both NULL, all rows are returned.
+      Otherwise, page_size rows are returned at offset page_offset.
+    mode: read
+    serviceName: index
+    args: parent_in text, name_in text, page_size_in integer, page_offset_in integer
+    returns: table(parent text, name text, expires timestamptz)
+    body: |-
+      begin
+        return query
+        select
+          index_namespaces.parent,
+          index_namespaces.name,
+          index_namespaces.expires
+        from index_namespaces
+        where
+          (index_namespaces.parent = parent_in or parent_in is null) and
+          (index_namespaces.name = name_in or name_in is null) and
```

Comment:
If we do not allow parent to be null then I think this defeats the purpose of a `get_index_namespaces` since it will only be able to return at most one entry.

Reply:
The primary key is (parent, name), so a call with a parent but no name would get all children of that parent.  But a call `get_index_namespaces(null, null, null, null)` would be a full table scan.  Anyway, this is a minor issue, since the scan is paginated and not exposed via the API.